### PR TITLE
textures: Refactor out for Texture/Depth FormatFromPixelFormat.

### PR DESCRIPTION
**Pull request by bunnei**
- As a plus, this allows us to add render target formats that do not have a corresponding texture format.

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 1]
```diff
@@ -183,6 +183,21 @@ MathUtil::Rectangle<u32> SurfaceParams::GetRect() const {
     return {0, actual_height, width, 0};
 }
 
+/// Returns true if the specified PixelFormat is a BCn format, e.g. DTX or DXN
```
Comment (MoochMcGee) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: *DXT

Comment (bunnei) [reply src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: ah good catch-ty

**Comment by MoochMcGee** [comment]
You should probably put the tile size calculations in the debugger too, but other than that, lgtm!!

**Comment by bunnei** [comment]
@MoochMcGee I did not want to duplicate the code because I'm not sure anyone uses this feature. If anyone is using it, they can update it - otherwise, I might remove it.

**Pull request merged**